term/fhir-term/src/main/java/com/ibm/fhir/term/util/CodeSystemSupport.java

```diff
@@ -232,6 +247,124 @@ public static CodeSystem getCodeSystem(java.lang.String url) {
         return FHIRRegistry.getInstance().getResource(url, CodeSystem.class);
     }
 
+    /**
+     * @param codeSystem
+     * @return
+     *    true if the passed codeSystem has a property-based concept hierarchy
+     */
+    public static boolean hasPropertyHierarchy(CodeSystem codeSystem) {
+        for (CodeSystem.Property p : codeSystem.getProperty()) {
+            Uri uri = p.getUri();
+            if (uri != null && PARENT_CHILD_PROPS.contains(uri.getValue())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Convert the passed codeSystem into a "simple" CodeSystem...one where all concept hierarchy is expressed
```

Comment:
I think to understand this fully I'd need some additional documentation showing the graphs with an example.

Reply:
Yes, it shouldbe be highly cachable.  However, my plan was to keep it simple for now.  There's very few property-based codesystems in the wild and we've already precomputed most (all?) of the valueset expansions for the ones with binding strength "extensible" or "required" in the core spec.